From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 27 Mar 2025 16:00:11 +0000
Subject: [PATCH] Fix application activate

Calling start() methods in activate() leads to crash on second activation.
Call them from startup() instead, and just present the window when the
application is launched second time.
---
 src/simple-scan.vala | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/simple-scan.vala b/src/simple-scan.vala
index cedbe060c80c..6e4a8f1a8c17 100644
--- a/src/simple-scan.vala
+++ b/src/simple-scan.vala
@@ -86,13 +86,15 @@ public class SimpleScan : Adw.Application
             app.set_scan_devices (device_list);
             app.set_selected_device (default_device.name);
         }
+
+        app.start ();
+        scanner.start ();
     }
 
     public override void activate ()
     {
         base.activate ();
-        app.start ();
-        scanner.start ();
+        app.present ();
     }
 
     public override void shutdown ()
